Fixed issue where PaginatedDataTable would not fill the width of its containing Card

bkonyi opened the pull request:
## Description
Fixed issue where PaginatedDataTable would not fill the width of its containing Card

**Before**
![before](https://user-images.githubusercontent.com/24210656/71297199-5a194380-2337-11ea-85d1-ab2e4ceec39e.png)
**After**
![after](https://user-images.githubusercontent.com/24210656/71297200-5be30700-2337-11ea-8ada-a7e06508ea89.png)

## Related Issues

- flutter/flutter#40081

bkonyi committed:
Fixed issue where PaginatedDataTable would not fill the width of its
containing Card

bkonyi commented:
The following test is failing, but I'm not exactly sure how to address the failure:

```
00:07 +6: PaginatedDataTable custom horizontal padding - checkbox                                                                                                                                                                                                                          
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════                                         
The following TestFailure object was thrown running a test:                                                                                  
  Expected: <10.0>                      
  Actual: <18.0>  
```

Adjusting the `_customHorizontalMargin` to 18 seems to result in the actual value changing to something larger.

Hixie commented on packages/flutter/lib/src/material/paginated_data_table.dart:
```diff
@@ -421,73 +421,78 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
+          ],
+        ),
       ),
     );
```
you should be able to fix this just by wrapping the Column in a `SizedBox.expand`.

bkonyi replied:
That seems to result in an unconstrained height which the `SingleChildScrollView` doesn't like:

```
════════ Exception caught by rendering library ═════════════════════════════════
The following assertion was thrown during performLayout():
BoxConstraints forces an infinite height.

These invalid constraints were provided to RenderFlex's layout() function by the following function, which probably computed the invalid constraints in question:
  RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:264:13)
The offending constraints were: BoxConstraints(w=1359.8, h=Infinity)
The relevant error-causing widget was
    PaginatedDataTable 
lib/…/flutter/network_screen.dart:233
When the exception was thrown, this was the stack
#0      BoxConstraints.debugAssertIsValid.<anonymous closure>.throwError 
package:flutter/…/rendering/box.dart:514
#1      BoxConstraints.debugAssertIsValid.<anonymous closure> 
package:flutter/…/rendering/box.dart:558
#2      BoxConstraints.debugAssertIsValid 
package:flutter/…/rendering/box.dart:562
#3      RenderObject.layout 
package:flutter/…/rendering/object.dart:1668
#4      RenderConstrainedBox.performLayout 
...
```

```
════════ Exception caught by rendering library ═════════════════════════════════
RenderBox was not laid out: RenderSemanticsAnnotations#8f706 relayoutBoundary=up1 NEEDS-PAINT
'package:flutter/src/rendering/box.dart':
Failed assertion: line 1691 pos 12: 'hasSize'
The relevant error-causing widget was
    SingleChildScrollView 
```

dnfield committed:
.. (#48263)

bkonyi committed:
Merge branch 'master' into fill_paginated_data_table_card

Hixie commented on packages/flutter/lib/src/material/paginated_data_table.dart:
```diff
@@ -426,74 +426,79 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
-                      mainAxisAlignment: MainAxisAlignment.end,
-                      children: headerWidgets,
+    return LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) => Card(
```
trivial nit: indentation

Hixie commented on packages/flutter/lib/src/material/paginated_data_table.dart:
```diff
@@ -426,74 +426,79 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
+                  opacity: 0.54
+                ),
+                child: Container(
+                  height: 56.0,
```
this looks like it won't handle text zoom correctly, fwiw

bkonyi replied:
Filed https://github.com/flutter/flutter/issues/48522

Hixie commented:
tests?

jmagman committed:
Add @christopherfujino as owner of dev/ci (docker) (#48261)

Commit:
update gitignore and xcode changes to macOS project for flutter gallery (#48251)

Commit:
[flutter_tool] Cache the path context in the flutter tool (#48250)

recomputing the path context in the getter violates flutter repo style guide and adds a small but measurable overhead to all path operations

mdebbar committed:
Registering multiple transformed events should work (#48149)

dnfield committed:
Make analyze once test not depend on test order or flutter create command (#48003)

dnfield committed:
Fix state in test (#48269)

Commit:
[flutter_tool] Enable globals technical debt benchmark (#48266)

a14n committed:
enable lint prefer_final_in_for_each (#47724)

Piinks committed:
SliverFadeTransition (#45950)

srawlins committed:
Mark unused but desired private constructors as such (#46200)

Piinks committed:
Refactor SliverFillRemaining (#47379)

zanderso committed:
Esarbanis flutter run help (#48314)

* Reword flutter run help screen.

* As per Hixie request, added Efthymios Sarmpanis to the AUTHORS file

* fix test

Co-authored-by: Efthymis Sarmpanis <e.sarbanis@gmail.com>

wise86-android committed:
[issue 7293] add test for rettangle box decoration with boarder radius (#47915)

srawlins committed:
Remove unused public static members of private classes/enums (#48218)

Commit:
[flutter_tools]web]Add support for --csp mode to build web (#48319)

dnfield committed:
analytics ordering issue (#48278)

zanderso committed:
Revert "Esarbanis flutter run help (#48314)" (#48351)

This reverts commit 866fa64dceab567750fb669f3a8f05b22e7e9342.

creativecreatorormaybenot committed:
Document reasoning for why CustomMultiChildLayout size can't depend on children and how to do it (#47797)

engine-flutter-autoroll committed:
Roll engine 3851981b863a..3f52888b3bcf (7 commits) (#48365)


https://github.com/flutter/engine/compare/3851981b863a...3f52888b3bcf

git log 3851981b863a..3f52888b3bcf --first-parent --oneline
2020-01-07 skia-flutter-autoroll@skia.org Roll src/third_party/skia d445e2b6fcf1..91e0d7526944 (44 commits) (flutter/engine#15232)
2020-01-07 dnfield@google.com Font subset (flutter/engine#14828)
2020-01-06 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from MwGkB... to xYmDF... (flutter/engine#15145)
2020-01-06 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/mac-amd64 from Kom_K... to nbeLe... (flutter/engine#15149)
2020-01-06 skia-flutter-autoroll@skia.org Roll src/third_party/skia 6791ee2f8059..d445e2b6fcf1 (1 commits) (flutter/engine#15148)
2020-01-06 skia-flutter-autoroll@skia.org Roll src/third_party/dart 76e3faa9824e..bebc7d3af5bd (1 commits) (flutter/engine#15146)
2020-01-04 skia-flutter-autoroll@skia.org Roll src/third_party/dart 2f57602411b3..76e3faa9824e (37 commits) (flutter/engine#15143)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

aam committed:
Revert "Document reasoning for why CustomMultiChildLayout size can't depend on children and how to do it (#47797)" (#48369)

This reverts commit de2514c0e2631e33996d131e8743fbcb7bbd57c4 as it breaks analyze-linux and web_tests-0-linux checks.

a14n committed:
use yield* (#48307)

aam committed:
Cleanup accidental use of global fs in test. (#48358)

This is follow-up to https://github.com/flutter/flutter/commit/ee7a37f1d3f12ca815d755a557f16765942897a4.

engine-flutter-autoroll committed:
Roll engine 3f52888b3bcf..a50f1ef56a05 (4 commits) (#48382)


https://github.com/flutter/engine/compare/3f52888b3bcf...a50f1ef56a05

git log 3f52888b3bcf..a50f1ef56a05 --first-parent --oneline
2020-01-08 gw280@google.com Migrate flutter_runner from flutter_runner::{Thread,Loop} to fml::{Thread,MessageLoop} (flutter/engine#15118)
2020-01-07 ferhat@gmail.com Refactor BitmapCanvas, lazily allocate canvas, fix image composition bug. (flutter/engine#15153)
2020-01-07 ferhat@gmail.com Recover when browser throws on ImageElement.decode due to too many images (flutter/engine#15160)
2020-01-07 garyq@google.com Fix RectHeightStyle::kMax ascent computation bug (flutter/engine#15106)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

Commit:
gen_l10n Number Format Handling (#47706)

dnfield committed:
Don't use local file system in devfs test (#48367)

dnfield committed:
StackTrace parser, fix assertion error message (#48343)

gspencergoog committed:
Add CatmullRomCurve and CatmullRomSpline (#47547)

This adds CatmullRomCurve animation curve, and a CatmullRomSpline, which is what it uses to do interpolation.

This allows us to create animation curves which can smoothly interpolate the values given to the curve.

Since I've introduced a 2D spline curve, I also created a Curve2D base class for such parametric curves.

engine-flutter-autoroll committed:
Roll engine a50f1ef56a05..4e9e69d2b392 (4 commits) (#48394)


https://github.com/flutter/engine/compare/a50f1ef56a05...4e9e69d2b392

git log a50f1ef56a05..4e9e69d2b392 --first-parent --oneline
2020-01-08 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/mac-amd64 from nbeLe... to gL-XG... (flutter/engine#15253)
2020-01-08 stuartmorgan@google.com Add plugin_registrar_windows.h to published wrapper (flutter/engine#15265)
2020-01-08 skia-flutter-autoroll@skia.org Roll src/third_party/dart bebc7d3af5bd..c0ca187f2699 (37 commits) (flutter/engine#15252)
2020-01-08 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from xYmDF... to oNKzr... (flutter/engine#15257)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

a14n committed:
enable lint prefer_for_elements_to_map_fromIterable (#47726)

engine-flutter-autoroll committed:
f001ea29f adding windows platform to felt. fixing signal (flutter/engine#15111) (#48422)


https://github.com/flutter/engine/compare/4e9e69d2b392...f001ea29f1b9

git log 4e9e69d2b392..f001ea29f1b9 --first-parent --oneline
2020-01-08 nurhan@google.com adding windows platform to felt. fixing signal (flutter/engine#15111)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

iskakaushik committed:
[fuchsia] Create an entry-point script to run driver tests on Fuchsia (#48359)

* [fuchsia] Create an entry-point script to run driver tests on Fuchsia

This script currently only paves the device. This will soon run a
driver test on a NUC in LUCI.

* reference the engine script and fix comment

ditman committed:
Change video_demo slightly so it works on web. (#46719)

Commit:
Reorganize tests in l10n tool (#47704)

Commit:
Update first-run information URLs (#48154)

marius-meissner committed:
Flexible padding (header height) for expanded panels (#47951)

* Added property for expanded padding of expansion panel for solving #24071

* #24071: Renamed default const name for being more descriptive

* #24071: Improved comment of property expandedPadding

* #24071: Improved test by expanding the expansion panel by header tap + precise assertion

* #24071: Renamed property to be more descriptive

* #24071: Adjusted property comment and added missing trailing commas

* #24071: Improved test by checking real exact header size

* #24071: Added test for default header padding

* #24071: Improved comment of expandedHeaderPadding + code style adjustments

* #24071: Added missing trailing comma

Commit:
[flutter_tool] Refactor Logger and Terminal to (mostly) no longer depend directly on context (#47269)

chingjun committed:
Rename conditional imported files to match convention (#48356)

stuartmorgan-g committed:
Check for desktop project files before building (#48350)

Moves the checks that projects have been configured for desktop to a lower level, where they will cover more codepaths (e.g., 'run'), and improves them to check for native build projects, rather than just directories, to catch cases where the directory exists (e.g., due to accidental creation of generated files).

Also adds links to the error messages pointing to instructions on adding desktop support to a project.

Fixes #47145

Commit:
Update Dart diagram (#48426)

engine-flutter-autoroll committed:
Roll engine f001ea29f1b9..46adf73d5128 (5 commits) (#48437)


https://github.com/flutter/engine/compare/f001ea29f1b9...46adf73d5128

git log f001ea29f1b9..46adf73d5128 --first-parent --oneline
2020-01-08 30870216+gaaclarke@users.noreply.github.com Made it so you can specify the old gen heap size. (flutter/engine#15259)
2020-01-08 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/mac-amd64 from gL-XG... to rTJJV... (flutter/engine#15308)
2020-01-08 xster@google.com Add a deprecation javadoc note to the old FlutterActivity (flutter/engine#15156)
2020-01-08 31859944+LongCatIsLooong@users.noreply.github.com Bump simulator version in IosUnitTests & scenario app in preparation for luci xcode 11 migration (flutter/engine#15154)
2020-01-08 skia-flutter-autoroll@skia.org Roll src/third_party/skia 91e0d7526944..f811fc331a14 (32 commits) (flutter/engine#15306)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC garyq@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

Sunbreak committed:
Add optional `endian` argument for WriteBuffer/ReadBuffer (#46661)

zanderso committed:
[flutter_tool] Don't use context in ProcessUtils (#48444)

dkwingsmt committed:
Add test of scrolling competition (#47492)

gspencergoog committed:
Reverse the sense of the terms snippet and sample. (#48254)

zanderso committed:
Revert "Roll engine f001ea29f1b9..46adf73d5128 (5 commits) (#48437)" (#48504)

This reverts commit 945f206b5e0eb83b3736ef7a5253ef12e32f630e.

Co-authored-by: LongCatIsLooong <31859944+LongCatIsLooong@users.noreply.github.com>

Piinks committed:
Gold Pre-submit flow for contributors without permissions (#47551)

christopherfujino committed:
Bump excon from 0.67.0 to 0.71.0 for ci on mac & linux (#48268)

zanderso committed:
Reland: Reword flutter run help screen. (#48423)

a14n committed:
format operator== according to flutter style guide (#48449)

creativecreatorormaybenot committed:
Document reasoning for why CustomMultiChildLayout size can't depend on children and how to do it (#48451)

Commit:
Implement reverseTransitionDuration for TransitionRoute (#48274)

* Implement reverseTransitionDuration in TransitionRoute

gspencergoog committed:
Simplify the Shortcuts widget diagnostic output (#48265)

This simplifies the diagnostic output for the Shortcuts widget so that if a debugLabel is supplied, then that is printed instead of the full list of shortcut keys in the map. Also, the output of the shortcut map is simplified to have the list of keys presented in more readable text.

LongCatIsLooong committed:
Add a diagnostic message for malformed imports in analyze.dart (#38355)

bkonyi commented:
Somehow managed to royally screw up the history here... will close and create a new PR.
